Automattic/vip-go-mu-plugins
# Add Debug Mode

Allows the VIP team to enter into Debug Mode when logged out to allow for faster, easier debugging of production-level issues.

- To enter: `?a8c-debug=true`
- To leave: `?a8c-debug=false` (or click on the debug flag)

When entering Debug Mode, the VIP nocache cookie (`vip-go-cb`) is set as well as a `a8c-debug` cookie. Proxy must be enabled at all times. All future requests are uncached. The Mode is enabled for two hours.

When in Debug Mode, Query Monitor is accessible for examining details about the request.

<img width="836" alt="2019-03-02 at 12 52 am" src="https://user-images.githubusercontent.com/86105/53677913-86b77e00-3c85-11e9-87de-f346ea1808d5.png">

Based off the work done by @shantanu2704 in #1030.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Make sure your traffic is not proxied.
1. When logged out, go to a page on the site.
1. Add `?a8c-debug=true` to the URL.
1. Verify you get an error

Then:

1. Proxy your traffic
1. Add `?a8c-debug=true` to the URL.
1. Verify that you get redirected back without the `a8c-debug` param.
1. Verify that you have the vip-go-cb cookie and the a8c-debug cookie.
1. Verify that the `A8C Debug` flag is visible on the bottom left.
1. Verify that the Query Monitor is accessible.

Then:

1. Turn off proxy and reload the page.
1. Verify that the flag and Query Monitor are no longer accessible.

And:

1. Verify that clicking on the flag removes that cookies and removes access to the flag and Query Monitor.
1. Verify that adding `?a8c-debug=false` redirects back and does the same.
